src/typegoose.ts

```diff
@@ -114,22 +114,14 @@ export function buildSchema<T, U extends NoParamConstructor<T>>(cl: U) {
   if (buildSchemas.get(cl.name)) {
     return buildSchemas.get(cl.name);
   }
-  let sch: mongoose.Schema<U>;
-  /** Parent Constructor */
-  let parentCtor = Object.getPrototypeOf(cl.prototype).constructor;
-  // iterate trough all parents
-  while (parentCtor && parentCtor.name !== 'Object') {
-    /* istanbul ignore next */
-    if (parentCtor.name === 'Typegoose') { // TODO: remove this if, if the Typegoose class gets removed [DEPRECATION]
-      deprecate(() => undefined, 'The Typegoose Class is deprecated, please try to remove it')();
-    }
-    // extend schema
-    sch = _buildSchema(parentCtor, sch);
-    // set next parent
-    parentCtor = Object.getPrototypeOf(parentCtor.prototype).constructor;
-  }
-  // get schema of current model
-  sch = _buildSchema(cl, sch);
+  const {schemaOptions = {}}: IModelOptions = Reflect.getMetadata(DecoratorKeys.ModelOptions, cl) || {};
+
+  const sch = _buildSchema(cl, schemaOptions);
+
+  Object.entries(schemaOptions)
```

Comment:
why does it need to be applied after the schema is build? isnt it already applied in _buildSchema?

Reply:
This is needed because the @prop decorators are executed before the getModelForClass method. This enhancement is covered by the test https://github.com/hasezoey/typegoose/pull/21/files#diff-ea520afea15a8887a0395e188d12c020R16


The root cause which makes this code necessary is in the schema.ts file (https://github.com/B-Stefan/typegoose/blob/r6/inheritance-upgrade/src/internal/schema.ts#L62)

The current execution order looks like this: 

1. Evaluation of @prop decorators 
2. One @prop decorator type is a `Garage` class 
3. It will hit the line (https://github.com/B-Stefan/typegoose/blob/r6/inheritance-upgrade/src/internal/schema.ts#L62) and create a new schema (without the options)
4. Evaluation of @modelOptions decorators will set metadata for the schema options
5. At some point somebody calls `buildSchema(Garage)`
6. It won't create a new Schema (https://github.com/B-Stefan/typegoose/blob/r6/inheritance-upgrade/src/internal/schema.ts#L62) because it already exists.
7. Options will be not set to the class.

Reply:
ok i think i now understood what you mean here, would it be better if we cache calls to prop & execture them when buildSchema is called?

Reply:
@hasezoey  Good idea! Thought about this as well and I think this will solve the problem. 
But on the other hand, it would add increase the complexity of the schema build process. 
Should we create some kind of improvement issue for this?

Reply:
yes, would be good, and i think too much for v6.0.0